### PR TITLE
Feat/media viewer

### PR DESCRIPTION
**Pull request by henrikbossart**
## What does this pull request change?
Media viewer to view media objects (File)

Does this somewhat resemble what we want/need?

(I'll remove the video file before merge)

## Issues related to this change

Screenshot 2023-11-15 at 09 40 12

### CODE REVIEW

File: example/app/data/DemoDataSource/apps/ExampleApplication/blueprints/MediaObject.blueprint.json
Comment (soofstad) [inline example/app/data/DemoDataSource/apps/ExampleApplication/blueprints/MediaObject.blueprint.json:1]: Sure we need this wrapper for the "File"?
We shouldn't

File: example/app/data/DemoDataSource/apps/ExampleApplication/blueprints/ExampleApplication.blueprint.json
Comment (soofstad) [inline example/app/data/DemoDataSource/apps/ExampleApplication/blueprints/ExampleApplication.blueprint.json:1]: Could we add the test blueprint and entities to "DemoDataSource/plugins/media_viewer", so it follows the structure of the other plugins and their tests

File: example/app/data/DemoDataSource/recipes/apps/ExampleApplication/mediaObject.recipe.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "type": "CORE:RecipeLink",
+  "_blueprintPath_": "/apps/ExampleApplication/blueprints/MediaObject",
```
Comment (soofstad) [inline example/app/data/DemoDataSource/recipes/apps/ExampleApplication/mediaObject.recipe.json:3]: I'd imagine this recipe would be for the "File" type instead.

File: example/app/data/DemoDataSource/recipes/file.recipe.json [hunk 2]
```diff
@@ -23,6 +23,21 @@
...
+          "type": "PLUGINS:dm-core-plugins/view_selector/ViewSelectorItem",
+          "label": "View",
+          "viewConfig": {
+            "type": "CORE:InlineRecipeViewConfig",
+            "recipe": {
+              "name": "View",
+              "type": "CORE:UiRecipe",
+              "description": "view media",
+              "plugin": "@development-framework/dm-core-plugins/media-viewer"
+            },
```
Comment (soofstad) [inline example/app/data/DemoDataSource/recipes/file.recipe.json:36]: Should remove this "recipe", with the "ViewConfig" type. "media-viewer" would then be selected automatically.

File: packages/dm-core-plugins/src/mediaViewer/MediaContent.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+`
+
+const MetaPopoverButton = styled(Button)`
+  position: absolute;
+  right: 0;
+  top: 0;
+  z-index: 50;
+  color: #595959;
+  //filter: invert(1);
+  //mix-blend-mode: difference;
```
Comment (soofstad) [inline packages/dm-core-plugins/src/mediaViewer/MediaContent.tsx:31]:
```suggestion
```

File: packages/dm-core-plugins/src/mediaViewer/MediaContent.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+`
+
+export function MediaContent(props: MediaContentProps): ReactElement {
+  const { blobUrl, meta } = props
+  const [showMeta, setShowMeta] = useState(false)
+  const referenceElement = useRef()
+  return (
+    <>
+      <MediaWrapper>
+        {meta.filetype !== 'application/pdf' && (
```
Comment (soofstad) [inline packages/dm-core-plugins/src/mediaViewer/MediaContent.tsx:51]: Could this be a "select case" instead? Would that make it more readable?

File: packages/dm-core-plugins/src/mediaViewer/MediaViewerPlugin.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+          setBlobUrl(window.URL.createObjectURL(blob))
+        })
+        .catch((error: AxiosError<ErrorResponse>) => {
+          console.error(error)
+        })
+  }, [document])
+
+  if (documentError) throw new Error(JSON.stringify(documentError, null, 2))
+  if (isLoading || document === null) return <Loading />
+  if (document.data.type !== EBlueprint.FILE) return <>Error: Not File type</>
```
Comment (soofstad) [inline packages/dm-core-plugins/src/mediaViewer/MediaViewerPlugin.tsx:68]: This might as well throw an Error

**Review by soofstad — CHANGES_REQUESTED** [review]
Looks great!
A few comments on structure

File: example/app/data/DemoDataSource/apps/ExampleApplication/blueprints/MediaObject.blueprint.json
Comment (henrikbossart) [reply example/app/data/DemoDataSource/apps/ExampleApplication/blueprints/MediaObject.blueprint.json:1]: No, I don't think we need it. Forgot to write about it in the description, but I didn't manage to just add the plugin for the File, so I had to create a separate entity/BP/recipe for it.

**Comment by henrikbossart** [comment]
Not quite sure I'm doing the blueprint/recipe/entity thing correctly, so I'll grab one of you guys tomorrow :+1:

**Comment by henrikbossart** [comment]
@soofstad please take a look at my recent changes regarding blueprints etc. I've changed it so that the file is rendered by the new media viewer by default, without the viewselector

File: example/app/data/DemoDataSource/plugins/media_viewer/assets/pain.mov
Comment (soofstad) [inline example/app/data/DemoDataSource/plugins/media_viewer/assets/pain.mov:1]: Would be nice to keep some test video. But if you can find a bit smaller one? :sweat_smile:

**Review by soofstad — APPROVED** [review]
Looks good!

**Pull request merged**